### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by skermes**

### CODE REVIEW

File: README.txt [hunk 1]
```diff
@@ -1,5 +1,126 @@
-This is a README.txt file
+# Welcome to Timberwolf
 
-The style of the project can be compared by running:
+Timberwolf is an open-source, command-line service to aid in importing emails
+from Microsoft Exchange into Hadoop hbase. It will locate all emails within an
+exchange instance which it has access to and upload them into hbase. It also
+allows for periodic recynching between Exchange and hbase through periodic
```
Comment (skermes) [inline README.txt]: "resyncing"

File: README.txt [hunk 2]
```diff
@@ -1,5 +1,126 @@
-This is a README.txt file
+# Welcome to Timberwolf
 
-The style of the project can be compared by running:
+Timberwolf is an open-source, command-line service to aid in importing emails
+from Microsoft Exchange into Hadoop hbase. It will locate all emails within an
```
Comment (skermes) [inline README.txt]: The convention seems to be to capitalize like "HBase".  We should probably stick to that.

**Comment by skermes** [comment]
Github doesn't seem to be picking up the markdown syntax automatically.  Try renaming README to README.md (as in https://github.com/github/markup)

File: README.txt [hunk 3]
```diff
@@ -1,5 +1,126 @@
...
+* Apache Maven
+
+To compile Timberwolf, run `mvn compile`.
+To build a distribution, run `mvn package`.
+
+The final Flume distribution artifacts will be in $project/target/.
+
+## Documentation
+
+We don't currently publish documentation.
```
Comment (skermes) [inline README.txt]: Maybe just leave this section out, then?  Or mention that they could build javadocs if they wanted?

File: README.txt [hunk 4]
```diff
@@ -1,5 +1,126 @@
...
+
+## Contact us
+
+_Not yet._
+
+## Bug and Issue tracker.
+
+_None._
```
Comment (skermes) [inline README.txt]: Underscores are for things that are pending other things, like having a website and functioning public JIRA?

File: README.txt [hunk 5]
```diff
@@ -1,5 +1,126 @@
...
+        --domain ${yourDomain}
+        --hbase-clientport ${portYourHbaseListensOn} \
+        --hbase-quorum ${yourHbaseUrl} \
+        --hbase-table ${resultingTable}
+
+## Contributing
+
+Timberwolf is an open-source project. Anyone can contribute to Timberwolf by
+submitting a patch (MORE DATA NEEDED HERE)
+
```
Comment (skermes) [inline README.txt]: Yeah, this feels a little gratuitous.  It's generally assumed that anything in a public Github repo is fair game for forking and pull-requesting.

File: README.txt [hunk 6]
```diff
@@ -1,5 +1,126 @@
...
+### Coding Conventions
+
+The Timberwolf code works off of the java sun coding conventions with a few
+minor exceptions. Curly braces should always be on their own line and imports
+should be ordered alphabetically and broken up by groups.
+
+You can run "mvn checkstyle:checkstyle" in order to confirm whether your
+additions conform to the coding convention.
```
Comment (skermes) [inline README.txt]: Have people other than me used the test/checkstyle/findbugs scripts I wrote?  Would it worth putting those on github (either under my account or riparian's) and pointing people to them as a friendlier alternative to raw static analysis?

File: README.txt [hunk 7]
```diff
@@ -1,5 +1,126 @@
-This is a README.txt file
+# Welcome to Timberwolf
 
-The style of the project can be compared by running:
+Timberwolf is an open-source, command-line service to aid in importing emails
+from Microsoft Exchange into Hadoop hbase. It will locate all emails within an
+exchange instance which it has access to and upload them into hbase. It also
+allows for periodic recynching between Exchange and hbase through periodic
+executions of the tool.
```
Comment (skermes) [inline README.txt]: This sentence feels super awkward.  Maybe something like "By keeping track of when it was last run, Timberwolf can periodically update HBase with new messages" or something

File: README.txt [hunk 8]
```diff
@@ -1,5 +1,126 @@
...
+  intend to use for testing. Likely, you can use the same configuration you
+  use for runtime, which is "Timberwolf".
+
+"testing.properties.example" is a sample file you can use to get you started.
+Just rename the properties appropriately and rename the file to
+"testing.properties". All of these properties directly correspond to command
+line arguments that Timberwolf accepts.
+
+Although the integrated tests run against exchange and hbase, they will not
+effect any existing data nor will they leave any testing data in hbase.
```
Comment (skermes) [inline README.txt]: "affect", not "effect"

File: README.txt (hunk 2 above)
Comment (ScottDugas) [inline README.txt]: The hbase has "Apache hbase" as the title, not "Hadoop hbase" so I would put the former
http://hbase.apache.org/

File: README.txt (hunk 7 above)
Comment (ScottDugas) [inline README.txt]: I'll second Sean's wording

File: README.txt [hunk 9]
```diff
@@ -1,5 +1,126 @@
...
+
+Compiling Timberwolf requires the following tools:
+
+* Java JDK 1.6
+* Apache Maven
+
+To compile Timberwolf, run `mvn compile`.
+To build a distribution, run `mvn package`.
+
+The final Flume distribution artifacts will be in $project/target/.
```
Comment (ScottDugas) [inline README.txt]: Flume? do you mean Timberwolf?

File: README.txt (hunk 5 above)
Comment (ScottDugas) [inline README.txt]: Do we want to commit on Pull Requests for submitting patches, if so we should say that

File: README.txt (hunk 6 above)
Comment (ScottDugas) [inline README.txt]: I was using them pretty extensively until I started on the mesasge pump

File: README.txt [hunk 10]
```diff
@@ -1,5 +1,126 @@
...
+additions conform to the coding convention.
+
+### Running the tests
+
+Before submitting a patch for approval, you should run the tests with:
+    mvn test
+
+There are a number of integrated tests that you can also run if you'd like,
+but they are ignored by default. In order to run the integrated tests, you
+must have a working instance of hbase and exchange to test against. And then
```
Comment (ScottDugas) [inline README.txt]: There are some integration tests that just work against hbase, without exchange, and they only need the hbase settings

File: README.txt [hunk 11]
```diff
@@ -1,5 +1,126 @@
...
+following properties:
+
+* *ZooKeeperQuorum*: This must specify the machine hosting your zookeeper
+  instance.
+* *ZooKeeperClientPort*= This specifies the port your zookeeper instance is
+  listening on.
+* *ExchangeURI* = This specifies the exchange web service you intend to hit.
+  It will most like end with ".../ews/exchange.asmx"
+* *LdapDomain* = This specifies the domain you're testing on.
+* *LdapConfigEntry* = This specifies your jaas login configuration entry you
```
Comment (ScottDugas) [inline README.txt]: "your jaas login" -> "the jaas login"

File: README.txt [hunk 12]
```diff
@@ -1,5 +1,126 @@
...
+  intend to use for testing. Likely, you can use the same configuration you
+  use for runtime, which is "Timberwolf".
```
Comment (ScottDugas) [inline README.txt]: This seems redundant, would it make more sense to just improve the comments in testing.properties.example and have the paragraph below

File: README.txt [hunk 13]
```diff
@@ -1,5 +1,126 @@
...
+
+"testing.properties.example" is a sample file you can use to get you started.
+Just rename the properties appropriately and rename the file to
+"testing.properties". All of these properties directly correspond to command
+line arguments that Timberwolf accepts.
```
Comment (ScottDugas) [inline README.txt]: LdapConfigEntry does not correspond to a command line argument; should it be an argument?

File: README.txt (hunk 8 above)
Comment (ScottDugas) [inline README.txt]: Should we be more precise here, we do create an hbase table, and then delete it.

File: README.txt (hunk 5 above)
Comment (ScottDugas) [inline README.txt]: A good one that I found (that would need some obvious changes for us):
Contributing
- Find something you would like to work on. For suggestions look for the `easy`, `medium` and `hard` tags in the issues
- Fork the project and do your work in a topic branch.
- Add shindo tests to prove your code works and run all the tests using `bundle exec rake`.
- Rebase your branch against geemus/fog to make sure everything is up to date.
- Commit your changes and send a pull request.

from https://github.com/fog/fog

Comment (skermes) [inline README.txt]: I rather like this one: https://github.com/github/help.github.com#readme

**Comment by skermes** [comment]
Regarding the length (Cameron and I talked about this briefly), I took a look at the top few trending and featured repos on github (https://github.com/explore) and they all have either no readme (bad developers!) or rather long readme's.  So I think we're fine.

File: README.txt (hunk 6 above)
Comment (ScottDugas) [inline README.txt]: Should we include our exception/logging conventions here?

**Pull request merged**